packages/near-membrane-base/src/membrane.ts

```diff
@@ -1160,15 +1197,19 @@ export function createMembraneMarshall() {
                 _shadowTarget: ShadowTarget,
                 key: string | symbol
             ): ReturnType<typeof Reflect.has> {
-                const { foreignTargetPointer } = this;
-                if (foreignCallableHasOwnProperty(foreignTargetPointer, key)) {
+                const trueOrProtoPointerOrNull =
+                    foreignCallableBatchGetPrototypeOfWhenHasNoOwnProperty(
+                        this.foreignTargetPointer,
+                        key
+                    );
+                if (trueOrProtoPointerOrNull === true) {
```

Comment:
☝️ Batched `foreignCallableHasOwnProperty` and `foreignCallableGetPrototypeOf` into one `foreignCallableBatchGetPrototypeOfWhenHasNoOwnProperty`.